internal/controller/siteconfig_builder.go

```diff
@@ -37,16 +40,113 @@ func NewSiteConfigBuilder(pLog logr.Logger) *SiteConfigBuilder {
 	return &scBuilder
 }
 
-// getConfigMap retrieves the configmap from cluster for a given TemplateRef
-func getConfigMap(ctx context.Context, c client.Client, templateRef v1alpha1.TemplateRef) (*corev1.ConfigMap, error) {
+func (scbuilder *SiteConfigBuilder) ProcessTemplates(ctx context.Context, c client.Client, siteConfig *v1alpha1.SiteConfig) ([]interface{}, error) {
+	scbuilder.Log.Info(fmt.Sprintf("Processing cluster-level templates for SiteConfig %s", siteConfig.Name))
 
-	cm := &corev1.ConfigMap{}
-	if err := c.Get(ctx, types.NamespacedName{
-		Name:      templateRef.Name,
-		Namespace: templateRef.Namespace,
-	}, cm); err != nil {
+	// Render cluster-level templates
+	siteManifests, err := scbuilder.renderTemplates(ctx, c, siteConfig, siteConfig.Spec.TemplateRefs, -1, siteConfig.Spec.SuppressedManifests)
```

Comment:
Consider creating a const for the "cluster" nodeId (-1).
Currently, it's a bit confusing.